src/EditorFeatures/Test/CommentSelection/ToggleLineCommentCommandHandlerTests.cs

```diff
@@ -0,0 +1,1084 @@
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.ToggleLineComment)]
+        public void AddComment_SelectionInsideCommentAtEndOfLine()
+        {
+            var markup =
+@"
+class C
+{
+    void M()
+    {
+        var i = 1; // A [|comment|].
+    }
+}";
+            var expected =
+@"
+class C
+{
+    void M()
+    {
+[|        //var i = 1; // A comment.|]
+    }
+}";
+
+            ToggleComment(markup, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.ToggleLineComment)]
+        public void AddComment_SelectionAroundCommentAtEndOfLine()
+        {
+            var markup =
+@"
+class C
+{
+    void M()
+    {
+        var i = 1; [|// A comment.|]
+    }
+}";
+            var expected =
+@"
+class C
+{
+    void M()
+    {
+[|        //var i = 1; // A comment.|]
+    }
+}";
+
+            ToggleComment(markup, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.ToggleLineComment)]
+        public void AddComment_SelectionOutsideCommentAtEndOfLine()
+        {
+            var markup =
+@"
+class C
+{
+    void M()
+    {
+        [|var i = 1; // A comment.|]
+    }
+}";
+            var expected =
+@"
+class C
+{
+    void M()
+    {
+[|        //var i = 1; // A comment.|]
+    }
+}";
+
+            ToggleComment(markup, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.ToggleLineComment)]
+        public void AddComment_CaretOutsideCommentAtEndOfLine()
+        {
+            var markup =
+@"
+class C
+{
+    void M()
+    {
+        var $$i = 1; // A comment.
+    }
+}";
+            var expected =
+@"
+class C
+{
+    void M()
+    {
+[|        //var i = 1; // A comment.|]
+    }
+}";
+
+            ToggleComment(markup, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.ToggleLineComment)]
+        public void AddComment_CaretInsideCommentAtEndOfLine()
+        {
+            var markup =
+@"
+class C
+{
+    void M()
+    {
+        var i = 1; // A $$comment.
+    }
+}";
+            var expected =
+@"
+class C
+{
+    void M()
+    {
+[|        //var i = 1; // A comment.|]
+    }
+}";
+
+            ToggleComment(markup, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.ToggleLineComment)]
+        public void AddComment_CommentMarkerInString()
+        {
+            var markup =
+@"
+class C
+{
+    void M()
+    {
+        [|string s = '\\';|]
+    }
+}";
+            var expected =
+@"
+class C
+{
+    void M()
+    {
+[|        //string s = '\\';|]
+    }
+}";
+
+            ToggleComment(markup, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.ToggleLineComment)]
+        public void AddComment_MultipleLinesSelected()
+        {
+            var markup =
+@"
+class C
+{
+    void M()
+    {
+        [|var i = 1;
+        var j = 2;|]
+    }
+}";
+            var expected =
+@"
+class C
+{
+    void M()
+    {
+[|        //var i = 1;
```

Comment:
so it doesn't do formatting?

Reply:
I'm not sure the question - it will get formatted, it's just in the markup it doesn't look formatted because the [| will get removed, so I have to account for that by adding extra spaces